networkdb/networkdb.go

```diff
@@ -487,16 +487,15 @@ func (nDB *NetworkDB) JoinNetwork(nid string) error {
 		},
 		RetransmitMult: 4,
 	}
-	nDB.networkNodes[nid] = append(nDB.networkNodes[nid], nDB.config.NodeName)
-	networkNodes := nDB.networkNodes[nid]
+	nDB.addNetworkNode(nid, nDB.config.NodeName)
 	nDB.Unlock()
 
 	if err := nDB.sendNetworkEvent(nid, NetworkEventTypeJoin, ltime); err != nil {
 		return fmt.Errorf("failed to send leave network event for %s: %v", nid, err)
 	}
 
 	logrus.Debugf("%s: joined network %s", nDB.config.NodeName, nid)
-	if _, err := nDB.bulkSync(networkNodes, true); err != nil {
+	if _, err := nDB.bulkSync(nDB.networkNodes[nid], true); err != nil {
```

Comment:
also not sure if nDB.networkNodes[nid] needs to be passed explicitly. I guess the bulkSync function can use networkNodes from the nDB object right ? wdyt ?